logstash-core/lib/logstash/agent.rb

```diff
@@ -77,6 +77,13 @@ def execute
   def register_pipeline(pipeline_id, settings)
     pipeline = create_pipeline(settings.merge(:pipeline_id => pipeline_id, :metric => metric))
     return unless pipeline.is_a?(LogStash::Pipeline)
+    if @auto_reload && pipeline.non_reloadable_plugins.any?
+      @logger.error(I18n.t("logstash.agent.non_reloadable_config_register"),
+                    :pipeline_id => pipeline_id,
+                    :configuration => pipeline.config_str,
```

Comment:
There's no need to dump the entire config here. It just hides the original error message